src/main/java/training/supportbank/Transaction.java

```diff
@@ -0,0 +1,20 @@
+package training.supportbank;
+
+import java.math.BigDecimal;
+import java.util.Date;
+
+public class Transaction {
+    final Date date;
+    final String from;
+    final String to;
+    final String narrative;
+    final BigDecimal amount;
+
+    public Transaction(Date tranDate, String tranFrom, String tranTo, String tranNarrative, BigDecimal tranAmount) {
```

Comment:
Java tip: it's idiomatic in Java to use the same names for the constructor parameter and the fields, so the lines below will be things like:

```
this.date = date;
```

Reply:
Done - makes sense.